pkg/token/server.go

```diff
@@ -218,70 +218,76 @@ func newHandlerFunc(d *daemon, timeout time.Duration) http.Handler {
+				if d.tokenType&mROLE_TOKEN != 0 {
+					rToken = d.roleTokenCache.Load(k)
+					if rToken == nil {
+						errMsg = fmt.Sprintf("domain[%s] role[%s] was not found in cache.", domain, role)
+					}
 				}
 			}
-		}
 
-		// check context cancelled
-		if r.Context().Err() != nil {
-			log.Warnf("Request context cancelled: URL[%s], domain[%s], role[%s], Err[%s]", r.URL.String(), domain, role, r.Context().Err().Error())
-			return
-		}
+			// check context cancelled
+			if r.Context().Err() != nil {
+				log.Warnf("Request context cancelled: URL[%s], domain[%s], role[%s], Err[%s]", r.URL.String(), domain, role, r.Context().Err().Error())
+				return
+			}
+
+			if len(errMsg) > 0 {
+				response, err := json.Marshal(map[string]string{"error": errMsg})
+				if err != nil {
+					log.Warnf("Error while preparing json response with: message[%s], error[%v]", errMsg, err)
+					w.WriteHeader(http.StatusInternalServerError)
+					return
+				}
+				errMsg = fmt.Sprintf("Error while handling request with: %s[%s] %s[%s], error[%s]", DOMAIN_HEADER, domain, ROLE_HEADER, role, errMsg)
+				log.Warnf(errMsg)
+				w.WriteHeader(http.StatusBadRequest)
+				io.WriteString(w, string(response))
+				return
+			}
 
-		if len(errMsg) > 0 {
-			response, err := json.Marshal(map[string]string{"error": errMsg})
+			resJSON := make(map[string]string, 2)
+			if aToken != nil {
+				at := aToken.Raw()
+				w.Header().Set("Authorization", "bearer "+at)
+				resJSON["accesstoken"] = at
+			}
+			if rToken != nil {
+				rt := rToken.Raw()
+				w.Header().Set(d.roleAuthHeader, rt)
+				resJSON["roletoken"] = rt
+			}
+			response, err := json.Marshal(resJSON)
 			if err != nil {
 				log.Warnf("Error while preparing json response with: message[%s], error[%v]", errMsg, err)
 				w.WriteHeader(http.StatusInternalServerError)
 				return
 			}
-			errMsg = fmt.Sprintf("Error while handling request with: %s[%s] %s[%s], error[%s]", DOMAIN_HEADER, domain, ROLE_HEADER, role, errMsg)
-			log.Warnf(errMsg)
-			w.WriteHeader(http.StatusBadRequest)
-			io.WriteString(w, string(response))
-			return
-		}
 
-		resJSON := make(map[string]string, 2)
-		if aToken != nil {
-			at := aToken.Raw()
-			w.Header().Set("Authorization", "bearer "+at)
-			resJSON["accesstoken"] = at
-		}
-		if rToken != nil {
-			rt := rToken.Raw()
-			w.Header().Set(d.roleAuthHeader, rt)
-			resJSON["roletoken"] = rt
-		}
-		response, err := json.Marshal(resJSON)
-		if err != nil {
-			log.Warnf("Error while preparing json response with: message[%s], error[%v]", errMsg, err)
-			w.WriteHeader(http.StatusInternalServerError)
+			log.Debugf("Returning %d for domain[%s], role[%s]", d.tokenType, domain, role)
+			io.WriteString(w, string(response))
 			return
 		}
 
-		log.Debugf("Returning %d for domain[%s], role[%s]", d.tokenType, domain, role)
-		io.WriteString(w, string(response))
+		w.WriteHeader(http.StatusNotFound)
+		io.WriteString(w, string("404 page not found"))
```

Comment:
I dont think you need this prolly. There could be many users with both rest api and envoy disabled.

Reply:
If you think you dont need this too, 
you dont have to do the 
```go
if d.tokenEnvoyAPI { ... }
```

and do 
```go
if !d.tokenEnvoyAPI {
  return
}
```

and not nest the tokenEnvoyAPI procedure (as it was)

Reply:
(even if you think you do need it, you can just do 
```go
if !d.tokenEnvoyAPI {
	w.WriteHeader(http.StatusNotFound)
	io.WriteString(w, string("404 page not found"))
	return
}
```

and leave the rest as it is.

Reply:
fixed: https://github.com/AthenZ/k8s-athenz-sia/pull/78/commits/d788a8a79292e2a0f617f05bb1bd204e5204fd24